### DISCUSSION TITLE
Deploying @auth/sveltekit to Cloudflare errors with [next-auth][error][UNTRUST_HOST_ERROR]

### DISCUSSION THREAD

**Discussion by alesvaupotic** [nextauthjs/next-auth#6071] [Help] [answered]
Hi!

Which ENV vars should I set to run from other provider than Vercel? The message is 

```json
"message": [
        "[next-auth][error][UNTRUST_HOST_ERROR]",
        "\nhttps://next-auth.js.org/errors#untrust_host_error",
        "Host must be trusted. URL was: https://skauth.pages.dev/api/auth/session",
        {
          "name": "UntrustedHost",
          "message": "Host must be trusted. URL was: https://skauth.pages.dev/api/auth/session",
          "stack": "UntrustedHost: Host must be trusted. URL was: https://skauth.pages.dev/api/auth/session\n    at AuthHandler (worker.mjs:7304:20)\n    at getSession (worker.mjs:7406:26)\n    at getSession._a.getSession (worker.mjs:7419:62)\n    at load (worker.mjs:9922:37)\n    at load_server_data (worker.mjs:7920:42)\n    at worker.mjs:8746:24"
        }
],
```

Vercel deploy works as expected.

**Comment by theetherGit** [comment, marked as answer]
AUTH_TRUST_HOST you can add this in your env file.

**Reply by alesvaupotic** [reply]
Works as expected, thanks!

**Reply by MeMartijn** [reply]
Hi! I'm running into the same issue - what value should AUTH_TRUST_HOST have? The domain name? Thanks!

**Reply by theetherGit** [reply]
Yes, you have to add your domain name if you get this error, it's only cause problem in dev right? So u can add your localhost or internal ip as host with http or https

**Reply by alesvaupotic** [reply]
It's actually a boolean as explained to me by the maintainers. 

https://authjs.dev/reference/sveltekit/#:~:text=When%20deploying%20your%20app%20outside%20Vercel%2C%20set%20the%20AUTH_TRUST_HOST%20variable%20to%20true%20for%20other%20hosting%20providers%20like%20Cloudflare%20Pages%20or%20Netlify.

**Reply by alesvaupotic** [reply]
More here:
https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/lib/types.ts#L215 
https://github.com/nextauthjs/next-auth/blob/main/packages/frameworks-sveltekit/src/lib/index.ts#L85

**Reply by MeMartijn** [reply]
@theetherGit it's actually just a problem on deployment, not locally. 
@alesvaupotic Setting the environment variable to "true" didn't fix it though, I'll try to deploy now with the URL to see if that fixes it. But looking at the code snippets you pointed at, that shouldn't make a difference...

**Reply by theetherGit** [reply]
@MeMartijn @alesvaupotic I'm using sveltekit and my answer is based on that.

export const AUTH_TRUST_HOST: string
As you can see AUTH_TRUST_HOST is a string in sveltekita-auth library https://github.com/nextauthjs/next-auth/blob/f7275c7527987e77d245e17467143d8b42a06f09/packages/frameworks-sveltekit/src/lib/index.ts#L103

I got this issue in dev preview only and in deployment it didn't bothered me.
For example you can take a look at this https://github.com/theetherGit/SvelteKit-v1-Guide it's working in deployment and don't have any issue related to it.

**Reply by suneelx** [reply]
> Yes, you have to add your domain name if you get this error, it's only cause problem in dev right? So u can add your localhost or internal ip as host with http or https

what if we have two domains like www.domain.com and domain.com?

**Reply by theetherGit** [reply]
About that even i don't know so can't help you with that.

**Reply by alesvaupotic** [reply]
In core it's a boolean, string in SvelteKit implementation for reading ENV only, I guess.

https://github.com/nextauthjs/next-auth/blob/f7275c7527987e77d245e17467143d8b42a06f09/packages/core/src/lib/types.ts#L223

**Reply by nmicun** [reply]
I have the same issue. Tried with addin in .env AUTH_TRUST_HOST="true", AUTH_TRUST_HOST="https://subdomain.domain" but nothing helps to me, always: 
![image](https://user-images.githubusercontent.com/56039142/213639475-b8b307a1-f07c-449a-8a6e-296bc6dc2d47.png)
Im using: 
"@auth/core": "^0.2.5",
"@auth/sveltekit": "^0.1.12",
"@next-auth/prisma-adapter": "^1.0.5",
problem is only on production

**Reply by suneelx** [reply]
> I have the same issue. Tried with addin in .env AUTH_TRUST_HOST="true", AUTH_TRUST_HOST="https://subdomain.domain" but nothing helps to me, always: ![image](https://user-images.githubusercontent.com/56039142/213639475-b8b307a1-f07c-449a-8a6e-296bc6dc2d47.png) Im using: "@auth/core": "^0.2.5", "@auth/sveltekit": "^0.1.12", "@next-auth/prisma-adapter": "^1.0.5", problem is only on production

@nmicun once please try AUTH_TRUST_HOST=subdomain.domain.com(e.g If I had .com.au, I would add .com.au)
also AUTH_TRUST_HOST=www.subdomain.domain.com

**Reply by nmicun** [reply]
Finally solved by adding trustHost: true
![image](https://user-images.githubusercontent.com/56039142/213649673-7fae91c7-8bdb-4c6e-8196-97149d29528e.png)
Is it safe?

**Reply by zimonitrome** [reply]
I also wonder if this is safe. But actually... I didn't even get it working with this option.

My application is a solid-start server with authjs.

**Reply by HectorMartinDama** [reply]
Thanks @nmicun

**Reply by mediocreaverage1** [reply]
@nmicun that was awesome. Thank you so much.

**Reply by xpluscal** [reply]
Setting "trustedHost" didn't work for me either. Using nodejs deployment and latest auth/core and auth/sveltekit,

**Reply by volstas** [reply]
Setting trustHost true in code worked. I don't understand why it doesn't work when set as environment variable Cloudlfare Pages, which is what all the docs said to do

**Reply by dtbndr** [reply]
adding environment variable didn't work. setting `trustedHost: true` worked for me for both cloudflare pages and localhost build preview.

**Reply by brunocleite** [reply]
@dtbndr Thanks, environment variable was not working for me either. Had to set `trustHost: true` in `auth.ts`
